### PR TITLE
Limit application id to 24 characters

### LINKED ISSUE #7938: Limit ApplicationId length to 24 characters
This is per design guidelines.

### PR DESCRIPTION
**Pull request by pakrym**
Fixes: https://github.com/Azure/azure-sdk-for-net/issues/7938

### CODE REVIEW

**Review by heaths — APPROVED** [review]
LGTM

File: sdk/core/Azure.Core/src/Pipeline/DiagnosticsOptions.cs [hunk 1]
```diff
@@ -48,7 +52,18 @@ internal DiagnosticsOptions()
         /// </summary>
         public IList<string> LoggingAllowedQueryParameters { get; }
 
-        public string? ApplicationId { get; set; }
+        public string? ApplicationId
```
Comment (MiYanni) [inline sdk/core/Azure.Core/src/Pipeline/DiagnosticsOptions.cs]: Would it be simpler to use this instead? https://docs.microsoft.com/en-us/dotnet/api/system.configuration.stringvalidatorattribute.maxlength?view=netframework-4.8

File: sdk/core/Azure.Core/src/Pipeline/DiagnosticsOptions.cs [hunk 2]
```diff
@@ -48,7 +52,18 @@ internal DiagnosticsOptions()
...
+        {
+            get => _applicationId;
+            set
+            {
+                if (value != null && value.Length > MaxApplicationIdLength)
+                {
+                    throw new ArgumentException($"{nameof(ApplicationId)} must be shorter than {MaxApplicationIdLength + 1} characters", nameof(value));
```
Comment (MiYanni) [inline sdk/core/Azure.Core/src/Pipeline/DiagnosticsOptions.cs]: Should use an `ArgumentOutOfRangeException`.

**Review by MiYanni — CHANGES_REQUESTED** [review]

File: sdk/core/Azure.Core/src/Pipeline/DiagnosticsOptions.cs (hunk 1 above)
Comment (pakrym) [reply sdk/core/Azure.Core/src/Pipeline/DiagnosticsOptions.cs]: That's for `System.Configuration` section definitions. It doesn't do any validation by itself.

Comment (MiYanni) [reply sdk/core/Azure.Core/src/Pipeline/DiagnosticsOptions.cs]: Oh. 😥

**Review by MiYanni — APPROVED** [review]

**Pull request merged**